src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -45,11 +48,26 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
             var param = F.Parameter(Parameters[0]);
             foreach (var field in ContainingType.GetFieldsToEmit())
             {
-                if (!field.IsStatic)
+                if (!field.IsStatic && field.ContainingType.Equals(ContainingType))
                 {
                     statements.Add(F.Assignment(F.Field(F.This(), field), F.Field(param, field)));
                 }
             }
         }
+
+        internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType)
+        {
+            foreach (var member in containingType.InstanceConstructors)
+            {
+                if (member is MethodSymbol { IsStatic: false, ParameterCount: 1, Arity: 0 } method &&
+                    method.Parameters[0].Type.Equals(containingType, TypeCompareKind.CLRSignatureCompareOptions) &&
```

Comment:
>CLRSignatureCompareOptions [](start = 85, length = 26)

I think we should use ```AllIgnoreOptions``` here, the ```CLRSignatureCompareOptions``` mode pays attention to aspects that language ignores.

Reply:
Okay, I think it makes sense to ignore custom modifiers since they are not mentioned in the language spec. But then we're introducing a problem of ambiguities which you mention below.
In such ambiguity cases, I'd propose that we simply consider we didn't find *the* copy constructor.

---
In reply to: [438764171](https://github.com/dotnet/roslyn/pull/45022#discussion_r438764171) [](ancestors = 438764171)

Reply:
> But then we're introducing a problem of ambiguities which you mention below.

The check you have doesn't help with the ambiguity because there are other places where modifiers can be in the signature.

---
In reply to: [438836863](https://github.com/dotnet/roslyn/pull/45022#discussion_r438836863) [](ancestors = 438836863,438764171)

Reply:
> I'd propose that we simply consider we didn't find the copy constructor.

I think it makes sense to try to disambiguate based on the count of modifiers. It should be simple to handle. Doesn't have to be addressed in this PR though

---
In reply to: [438860888](https://github.com/dotnet/roslyn/pull/45022#discussion_r438860888) [](ancestors = 438860888,438836863,438764171)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/45077](https://github.com/dotnet/roslyn/issues/45077) for custom modifiers scenarios

---
In reply to: [438862032](https://github.com/dotnet/roslyn/pull/45022#discussion_r438862032) [](ancestors = 438862032,438860888,438836863,438764171)